### ISSUE TITLE
RDS destruction fails with: Instance db-name is already being deleted

### ISSUE THREAD

**Issue by mildred** [terraform-providers/terraform-provider-aws#5622]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

- terraform v0.11.4
- aws provider version v1.27.0

### Description

When running `terraform destroy` with a RDS resources in the tfstate, terraform tries to delete it. It can happen that the aws-sdk-go delete instance request is returning with the following error:

> Validate Response rds/DeleteDBInstance failed, not retrying, error InvalidDBInstanceState: Instance db-qa-production-20180717110038-2-production is already being deleted. 
> status code: 400, request id: d5ba144e-d3f2-49c3-96ea-484dec0f78bb

This is caused by the SDK making a first request (that worked), having the HTTP connection closed before the SDK could receive the response. The SDK then retries, but the retry request is telling that the instance is already being deleted.

The correct behaviour in that case is probably to ignore the error because the action desired (deleting the RDS instance) is already in progress. We only have to wait for deletion to complete.

### Affected Resource(s)

* aws_db_instance

### Terraform Configuration Files

Nothing specific here

### Debug Output

```
2018/07/17 11:21:15 [DEBUG] [aws-sdk-go] DEBUG: Request rds/DeleteDBInstance Details: 
---[ REQUEST POST-SIGN ]----------------------------- 
POST / HTTP/1.1 
Host: rds.eu-west-1.amazonaws.com 
User-Agent: aws-sdk-go/1.13.28 (go1.10.2; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.4 
Content-Length: 131 
Content-Type: application/x-www-form-urlencoded; charset=utf-8 
X-Amz-Date: 20180717T112115Z 
Accept-Encoding: gzip 

Action=DeleteDBInstance&DBInstanceIdentifier=db-qa-production-20180717110038-2-production&SkipFinalSnapshot=true&Version=2014-10-31 
----------------------------------------------------- 
2018/07/17 11:21:27 [DEBUG] [aws-sdk-go] DEBUG: Send Request rds/DeleteDBInstance failed, will retry, error RequestError: send request failed 
caused by: Post https://rds.eu-west-1.amazonaws.com/: EOF 
2018/07/17 11:21:27 [DEBUG] [aws-sdk-go] DEBUG: Retrying Request rds/DeleteDBInstance, attempt 1 
2018/07/17 11:21:27 [DEBUG] [aws-sdk-go] DEBUG: Request rds/DeleteDBInstance Details: 
---[ REQUEST POST-SIGN ]----------------------------- 
POST / HTTP/1.1 
Host: rds.eu-west-1.amazonaws.com 
User-Agent: aws-sdk-go/1.13.28 (go1.10.2; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.4 
Content-Length: 131 
Content-Type: application/x-www-form-urlencoded; charset=utf-8 
X-Amz-Date: 20180717T112127Z 
Accept-Encoding: gzip 

Action=DeleteDBInstance&DBInstanceIdentifier=db-qa-production-20180717110038-2-production&SkipFinalSnapshot=true&Version=2014-10-31 
----------------------------------------------------- 
2018/07/17 11:21:39 [DEBUG] [aws-sdk-go] DEBUG: Send Request rds/DeleteDBInstance failed, will retry, error RequestError: send request failed 
caused by: Post https://rds.eu-west-1.amazonaws.com/: EOF 
2018/07/17 11:21:39 [DEBUG] [aws-sdk-go] DEBUG: Retrying Request rds/DeleteDBInstance, attempt 2 
2018/07/17 11:21:39 [DEBUG] [aws-sdk-go] DEBUG: Request rds/DeleteDBInstance Details: 
---[ REQUEST POST-SIGN ]----------------------------- 
POST / HTTP/1.1 
Host: rds.eu-west-1.amazonaws.com 
User-Agent: aws-sdk-go/1.13.28 (go1.10.2; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.4 
Content-Length: 131 
Content-Type: application/x-www-form-urlencoded; charset=utf-8 
X-Amz-Date: 20180717T112139Z 
Accept-Encoding: gzip 

Action=DeleteDBInstance&DBInstanceIdentifier=db-qa-production-20180717110038-2-production&SkipFinalSnapshot=true&Version=2014-10-31 
----------------------------------------------------- 
2018/07/17 11:21:51 [DEBUG] [aws-sdk-go] DEBUG: Send Request rds/DeleteDBInstance failed, will retry, error RequestError: send request failed 
caused by: Post https://rds.eu-west-1.amazonaws.com/: EOF 
2018/07/17 11:21:51 [DEBUG] [aws-sdk-go] DEBUG: Retrying Request rds/DeleteDBInstance, attempt 3 
2018/07/17 11:21:51 [DEBUG] [aws-sdk-go] DEBUG: Request rds/DeleteDBInstance Details: 
---[ REQUEST POST-SIGN ]----------------------------- 
POST / HTTP/1.1 
Host: rds.eu-west-1.amazonaws.com 
User-Agent: aws-sdk-go/1.13.28 (go1.10.2; linux; amd64) APN/1.0 HashiCorp/1.0 Terraform/0.11.4 
Content-Length: 131 
Content-Type: application/x-www-form-urlencoded; charset=utf-8 
X-Amz-Date: 20180717T112151Z 
Accept-Encoding: gzip 

Action=DeleteDBInstance&DBInstanceIdentifier=db-qa-production-20180717110038-2-production&SkipFinalSnapshot=true&Version=2014-10-31 
----------------------------------------------------- 
2018/07/17 11:21:52 [DEBUG] [aws-sdk-go] DEBUG: Response rds/DeleteDBInstance Details: 
---[ RESPONSE ]-------------------------------------- 
HTTP/1.1 400 Bad Request 
Connection: close 
Content-Length: 332 
Content-Type: text/xml 
Date: Tue, 17 Jul 2018 11:21:51 GMT 
X-Amzn-Requestid: d5ba144e-d3f2-49c3-96ea-484dec0f78bb 


----------------------------------------------------- 
2018/07/17 11:21:52 [DEBUG] [aws-sdk-go] <ErrorResponse xmlns="http://rds.amazonaws.com/doc/2014-10-31/"> 
<Error> 
<Type>Sender</Type> 
<Code>InvalidDBInstanceState</Code> 
<Message>Instance db-qa-production-20180717110038-2-production is already being deleted.</Message> 
</Error> 
<RequestId>d5ba144e-d3f2-49c3-96ea-484dec0f78bb</RequestId> 
</ErrorResponse> 
2018/07/17 11:21:52 [DEBUG] [aws-sdk-go] DEBUG: Validate Response rds/DeleteDBInstance failed, not retrying, error InvalidDBInstanceState: Instance db-qa-production-20180717110038-2-production is already being deleted. 
status code: 400, request id: d5ba144e-d3f2-49c3-96ea-484dec0f78bb
```

### Expected Behavior

To see no error, the database is deleted

### Actual Behavior

Terraform failed with an error

### Steps to Reproduce

Very difficult to

### Important Factoids

possibly caused by API rate limiting

### References

N/A

**Comment by bflad** [comment]
Hi @mildred 👋  Thanks for submitting this and sorry it gave you trouble.

Your timing is pretty good as I just ran into this while doing some extensive acceptance testing with that resource:

```
    Error: Error applying: 1 error occurred:
      * aws_db_instance.test (destroy): 1 error occurred:
      * aws_db_instance.test: InvalidDBInstanceState: Instance tf-acc-test-2488095917993051942 is already being deleted.
```

Since I'm already in the code neighborhood, I should be able to get this fixed up with acceptance testing shortly. 👍

**Comment by bflad** [comment]
Pull request submitted: #5624

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.33.0 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.33.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1330-august-22-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
